src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -907,13 +907,43 @@ uct_ib_mlx5_devx_obj_create(struct ibv_context *context, const void *in,
                             size_t inlen, void *out, size_t outlen,
                             char *msg_arg, ucs_log_level_t log_level);
 
-ucs_status_t
-uct_ib_mlx5_devx_obj_destroy(struct mlx5dv_devx_obj *obj, char *msg_arg);
+static inline ucs_status_t
+uct_ib_mlx5_devx_obj_destroy(struct mlx5dv_devx_obj *obj, char *msg_arg)
+{
+    int ret;
+
+    ret = mlx5dv_devx_obj_destroy(obj);
+    if (ret != 0) {
+        ucs_warn("mlx5dv_devx_obj_destroy(%s) failed: %m", msg_arg);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    return UCS_OK;
+}
+
+
+static inline ucs_status_t
+uct_ib_mlx5_devx_general_cmd(struct ibv_context *context,
+                             const void *in, size_t inlen,
+                             void *out, size_t outlen,
+                             char *msg_arg, int silent)
+{
+    ucs_log_level_t level = silent ? UCS_LOG_LEVEL_DEBUG : UCS_LOG_LEVEL_ERROR;
+    int ret;
+    unsigned syndrome;
+
+    ret = mlx5dv_devx_general_cmd(context, in, inlen, out, outlen);
+    if (ret != 0) {
+        syndrome = UCT_IB_MLX5DV_GET(general_obj_out_cmd_hdr, out, syndrome);
+        ucs_log(level,
+                "mlx5dv_devx_general_cmd(%s) failed on %s, syndrome 0x%x: %m",
+                msg_arg, ibv_get_device_name(context->device), syndrome);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    return UCS_OK;
+}
```

Comment:
does it need to be inlined? maybe move to `ib_mlx5.c`

Reply:
`src/uct/ib/rdmacm` needs them. By moving those functions to headers, UCT lib rdmacm does not depend on the new libuct_ib_mlx5.so at runtime, but only on libmlx5.so. This is to avoid bringing lib dependency for a pair of simple functions.

Reply:
tbh I tried that route first but this created many headaches wrt lib dependency as uct lib rdmacm is itself a module. If I recall one of them, rdmacm misses libuct_ib_mlx5 symbols, although they are loaded by libuct_ib.so code with dlopen(). There were other issues like also cannot directly link rdmacm to libuct_ib_mlx5.so, but I do not recall details exactly.

Since it's only two small functions, I decided to simplify like that. Else we could also reimplement those two simple functions to not even depend on libuct_ib_mlx5 headers.

Reply:
ok